Upload Attendance feature and Reports bug fixes

argunda1 opened the pull request:
This PR contains the following changes:
1) Upload attendance feature --> attendance file upload feature on workshop info page which automatically creates attendance records based on the UUIDs and also creates new teachers as needed
2) New report - teacher report --> teacher report added to Reports list, which lists all the teachers in the system irrespective of the department, along with the first login date/time for easy retrieval of latest teachers
3) pagination fix on department report --> fixed missing pagination setting and links on department list
4) Alphabetical order in Department dropdown --> Changed department order to be based on the department title instead of id
5) Edit Department - new page --> Added edit functionality for departments. Can be accessed from Department report. Shows fields for Old department, new department and STEM.
6) Move the content in departmentEdit.blade to departmentAddOrRemove.blade --> For Naming consistency
7) Replace "$t->department->title." with department checks and empty placeholders --> Handled empty department names on all the pages, as the upload attendance feature doesn't add department details for a new teacher.

argunda1 committed:
Changes for
1) Upload attendance feature
2) New report - teacher report
3) pagination fix on department report
4) Alphabetical order in Department dropdown
5) Edit Department - new page
6) Move the content in departmentEdit.blade to departmentAddOrRemove.blade
7) Replace "$t->department->title." with department checks and empty placeholders

trevordimartino reviewed (approved):
Great work here!  I haven't tested anything, but I know from working with you that you have tested it all extensively already.  😄 

The one question I have is whether you are truly deleting departments or just "soft deleting" them.  I can't quite tell from the code what's happening there, but if it's a "soft delete," then great job!

argunda1 commented:
Thanks for approving, Trevor!
I haven't modified anything on "delete" department functionality. I just added "edit" department functionality and moved the code from old 'departmentEdit' file to 'departmentAddOrRemove'.
